aten/src/ATen/native/TensorIterator.h

```diff
@@ -467,7 +468,16 @@ class CAFFE2_API TensorIteratorConfig final {
   //   These temporaries are then copied to the original outputs after
   //   the operation is performed (see cast_outputs()).
   // Setting this flag to true sets check_all_same_dtype_ to false.
-  TensorIteratorConfig& cast_common_dtype_to_outputs(const bool _cast_common_dtype_to_outputs);
+  //
+  // Note [special_case_bool_outputs]
```

Comment:
Is there a more distinct name we can use for this flag? TensorIterators have a few flags that handle special cases. Maybe something like `optimize_bool_outputs` would be clearer?

Reply:
That would probably be a better name. After talking to Mike, I opted to remove the flag altogether and add the equivalent logic directly into the function that creates the `comparison_op` TensorIterator. This seems cleaner, since business of avoiding a copy for bool outputs is specific to comparison/logical ops (and we don't need a new flag).